.azure-pipelines/Linux-CI.yml

```diff
@@ -37,22 +47,30 @@ jobs:
     inputs:
       createCustomEnvironment: true
       environmentName: 'py$(python.version)'
-      packageSpecs: 'python=$(python.version) protobuf'
+      packageSpecs: 'python=$(python.version)'
+      installOptions: '-c conda-forge protobuf'
 
   - script: |
       python -m pip install --upgrade pip
       sudo apt-get install -qq -o=Dpkg::Use-Pty=0 -y --no-install-recommends dos2unix
+      sudo apt-get install -y --no-install-recommends protobuf-compiler libprotoc-dev
```

Comment:
Is this needed? given you are adding it in conda on line 51

Reply:
It was needed to resolve CI errors, even with the conda-forge install. The CI was failing because it could not identify the "protobuf compiler" without that line.

Reply:
Referenced run: https://dev.azure.com/onnx-pipelines/onnx/_build/results?buildId=3188&view=logs&jobId=2105a913-e5a3-5f74-1c30-4f8b7a56f2db&j=2105a913-e5a3-5f74-1c30-4f8b7a56f2db&t=c2afa4d8-3a4c-5e2d-1542-4328675067bd

Reply:
Then you could run into protobuf mismatch problem. You'll use the protoc from the operating system(protobuf-compiler package), but libprotobuf from conda(based on your description). Otherwise you don't need two. 

If the could not identify the "protobuf compiler", please check if your PATH env variable has setup correctly. Please reference conda's document to see which env vars need be updated.

Reply:
I looked at your reference run. The protobuf package is not being installed in the conda environment that is why it cant find it.

Reply:
Ideally, protoc, libprotobuf and protobuf python package, all these three components should have the same version. But it seems ok if the python one has higher version than the other two.

Reply:
I've removed the two dependencies (libprotoc-dev and protobuf-compiler) and used a conda install -c conda-forge protobuf line in the script directly (like in the CI for onnxmltools -- https://github.com/onnx/onnxmltools/pull/207).

Reply:
It seems like the install error is a possible bug with azure pipelines' conda setup task. Thanks for the advice!

Reply:
Hi @askhade, @snnn   -- I can't seem to get the CIs to pass without these two dependencies. It's not finding them in the source package, even when I move the conda install -c conda-forge protobuf line in this block. Is it ok to install these dependencies as well?

Reply:
As snnn explained above this might create other issues because you will end up using protoc and libprotobuf from different sources. Can you check whether protobuf compiler was indeed installed? If yes then fixing the PATH should resolve this issue otherwise can you simply get rid of conda? can you use virtualenvv for env management?